flake8_idom_hooks/utils.py

```diff
@@ -28,12 +28,18 @@ def is_hook_def(node: ast.FunctionDef) -> bool:
 
 
 def is_component_def(node: ast.FunctionDef) -> bool:
-    return is_component_function_name(node.name)
-
-
-def is_component_function_name(name: str) -> bool:
-    return name[0].upper() == name[0] and "_" not in name
+    return any(
+        decorator.value.id == "idom" and decorator.attr == "component"
```

Comment:
This seems to require that the component be decorated specifically with `@idom.component`. It would be better if this more generally identified something of the form `@*.component` as a component. For example, all these should be valid:

```
@component
@idom.component
@idom.core.component.component
@x.y.z.component
```

This protects against the various ways the component decorator might be imported by a user.

Reply:
I've confirmed this works for both. Seems like the syntax tree automatically expands out since it understands the parent package is `idom`

Reply:
I'd have to test and see if `decorator.value.id,startswith("idom") and decorator.attr == "component"` would work.

Reply:
Supporting `idom.component` and `component` should be fine I think.